src/inmanta/server/services/orchestrationservice.py

```diff
@@ -586,3 +586,29 @@ async def desired_state_version_list(
         )
 
         return ReturnValueWithMeta(response=dtos, links=links if links else {}, metadata=vars(metadata))
+
+    @handle(methods_v2.promote_desired_state_version, env="tid")
+    async def promote_desired_state_version(
+        self,
+        env: data.Environment,
+        version: int,
+        trigger_method: Optional[PromoteTriggerMethod] = None,
+    ) -> None:
+        if trigger_method == PromoteTriggerMethod.push_incremental_deploy:
+            push = True
+            agent_trigger_method = const.AgentTriggerMethod.push_incremental_deploy
+        elif trigger_method == PromoteTriggerMethod.push_full_deploy:
+            push = True
+            agent_trigger_method = const.AgentTriggerMethod.push_full_deploy
+        elif trigger_method == PromoteTriggerMethod.no_push:
+            push = False
+            agent_trigger_method = None
+        else:
+            push = True
+            agent_trigger_method = None
+
+        status_code, result = await self.release_version(
+            env, version_id=version, push=push, agent_trigger_method=agent_trigger_method
+        )
+        if status_code == 404:
+            raise NotFound(result["message"])
```

Comment:
An `status_code` different from `200` and `404` will be translated to `200` by this endpoint. This is not desired I suppose?